plugin/completion.py

```diff
@@ -40,11 +41,100 @@
+            self.view.hide_popup()
+
+
 class LspResolveDocsCommand(sublime_plugin.TextCommand):
+    def run(self, edit: sublime.Edit, item) -> None:
+        detail = item.get('detail')
+        documentation = self.normalized_documentation(item)
+
+        # don't show the detail in the cooperate AC popup,
+        # if it is alread shown in the AC details filed.
+        self.is_detail_shown = bool(detail)
+
+        content = self.get_content(documentation, detail)
+        self.show_popup(content)
+
+        if not detail or not documentation:
+            # To make sure that the details or documentation fields don't exist
+            # we resove the completion item.
+            # If those fields appear after the item is resolved
+            # update the popup.
+            self.do_resolve(item)
+
+    def show_popup(self, content):
+        mdpopups.show_popup(
+            self.view,
+            content,
+            flags=sublime.COOPERATE_WITH_AUTO_COMPLETE,
+            max_width=480,
+            max_height=333,
+            allow_code_wrap=True
+        )
+
+    def update_popup(self, content):
+        mdpopups.update_popup(
+            self.view,
+            content
+        )
+
+    def do_resolve(self, item: dict) -> None:
+        session = session_for_view(self.view, 'completionProvider', self.view.sel()[0].begin())
+        if not session:
+            return
 
-    def run(self, edit: sublime.Edit) -> None:
-        self.view.show_popup('<div style="padding: 10px;">Showing documentation is under construction</div>',
-                             sublime.COOPERATE_WITH_AUTO_COMPLETE)
+        client = client_from_session(session)
+        if not client:
+            return
+
+        completion_provider = session.get_capability('completionProvider')
+        has_resolve_provider = completion_provider and completion_provider.get('resolveProvider', False)
+        if has_resolve_provider:
+            client.send_request(Request.resolveCompletionItem(item), self.handle_resolve_response)
+
+    def handle_resolve_response(self, item):
+        documentation = self.normalized_documentation(item)
+        detail = item.get('detail')
+
+        content = self.get_content(documentation, detail)
+        self.show_popup(content)
+
+    def get_content(self, documentation, detail):
+        content = ""
+        if detail and not self.is_detail_shown:
+            content += """<div class='highlight' style='margin: 10px'>
+                <p>{}</p>
+            </div>""".format(detail)
+
+        if documentation:
+            content += """<div style='margin: 10px'>
+                <p>{}</p>
+            </div>""".format(documentation)
+
+        return content
+
+    def normalized_documentation(self, item: Dict[str, Any]) -> str:
+        lsp_documentation = item.get("documentation")
+        if isinstance(lsp_documentation, str):
+            return mdpopups.md2html(self.view, lsp_documentation)
```

Comment:
Should this not be `html.escape(lsp_documentation)`? Why do you asssume that it is markdown?

Reply:
same reason, https://github.com/sublimelsp/LSP/pull/1022/files#r420379578

Reply:
With the example you've linked:
```
'documentation': {'value': "Specifies the width of the content area, padding area or border area (depending on 'box-sizing') of certain boxes.\n\nSyntax: <viewport-length>{1,2}\n\nMDN Reference: https://developer.mozilla.org/docs/Web/CSS/width", 'kind': 'plaintext'}
```
The link will become clickable but also the `<viewport-length>` will disappear as it will be parsed as HTML element. And it's not supposed to be. It's supposed to be shown verbatim.

(CSS server is just stupid. It just use markdown formatting but escaping stuff that is supposed to be escaped.)

Reply:
fixed with the minihtml function, https://github.com/sublimelsp/LSP/pull/1022/commits/8a803e0077b3eecaae56bf867155c3d9e7d96515#diff-f52730e9958cbd7da304f8aa67516d5cR179-R194